python/revng/ptml/common.py

```diff
@@ -78,22 +74,17 @@ def handle_file(
         return handle_multiple(parsed_yaml, func_one, func_many, filters)
 
 
-T = TypeVar("T")
-P = ParamSpec("P")
-
-
-def suppress_brokenpipe(func: Callable[P, T], *args: P.args, **kwargs: P.kwargs) -> Optional[T]:
-    """When running a program with a pipe, BrokenPipeError might be raised. This signals that the
-    output pipe was closed, which we expect. Suppressing the exception is not enough since it can
-    also happen at shutdown, which will trigger python's unraisable hook, to remedy this we
-    overwrite the default hook to ignore BrokenPipeError."""
-
-    def new_unraisablehook(arg):
-        if arg.exc_type != BrokenPipeError:
-            sys.__unraisablehook__(arg)
+def parse(raw: bytes, console=None, filters=[]):
```

Comment:
I added the types, or did you intended that I should have 2 implementations?

Reply:
Python supports function overloading for types, e.g.:
```
def parse(raw: bytes, console: None, filters: List[str]) -> str: ...
def parse(raw: bytes, console: Console, filters: List[str]) -> None: ...
def parse(raw: bytes, console: Console | None = None, filters: List[str] = []):
   <actual body>
```

Tells the typechecker that if `console` is `None` the return should be `str`, otherwise `None`